core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -93,6 +95,16 @@ static void toJson(Snapshot snapshot, JsonGenerator generator) throws IOExceptio
       generator.writeNumberField(SCHEMA_ID, snapshot.schemaId());
     }
 
+    if (snapshot.manifestListKeyMetadata() != null) {
+      generator.writeStringField(MANIFEST_LIST_KEY_METADATA, snapshot.manifestListKeyMetadata());
+    }
+
+    // TODO discuss: do we need to sign the size value? Or sign the whole snapshot?
```

Comment:
How would this attack work? Wouldn't the user also need the key to encrypt the replacement files? I thought we were storing the metadata.json key in the catalog so an attacker could replace everything but still not be able to trick a client using the catalog.